core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -1281,6 +1341,33 @@ private[spark] class MapOutputTrackerWorker(conf: SparkConf) extends MapOutputTr
     }
   }
 
+  override def getShufflePushMergerLocations(shuffleId: Int): Seq[BlockManagerId] = {
+    shufflePushMergerLocations.getOrElse(shuffleId, getMergerLocations(shuffleId))
+  }
+
+  private def getMergerLocations(shuffleId: Int): Seq[BlockManagerId] = {
+    val mergers = shufflePushMergerLocations.get(shuffleId).orNull
+    if (mergers == null) {
```

Comment:
nit:
This (get and null check) is not required - we are invoking this method only when `shufflePushMergerLocations` does not have an entry.